components/scream/src/mct_coupling/tests/scorpio_tests.cpp

```diff
@@ -0,0 +1,114 @@
+  register_variable(outfilename,"index_2d","test value for 2d field",3,vec_xyt, PIO_INT,"xyt-int");
+  register_variable(outfilename,"index_3d","test value for 3d field",4,vec_xyzt, PIO_INT,"xyzt-int");
+  // Finished with the initialization of variables in output file
+  eam_pio_enddef(outfilename);
+  // Create data to be written
+  std::array<Real,10> x_data;
+  std::array<Real, 5> y_data;
+  std::array<Real, 2> z_data;
+  std::array<Int,1> xdim = {10};
+  std::array<Int,1> ydim = {5};
+  std::array<Int,1> zdim = {3};
+  std::array<Int,1> dimlen_1d = {10};
+  std::array<Int,2> dimlen_2d = {5,10};
+  std::array<Int,3> dimlen_3d = {2,5,10};
+  ekat::util::md_array<Real,10>       test_data_1d;
+  ekat::util::md_array<Real, 5,10>    test_data_2d;
+  ekat::util::md_array<Real, 2, 5,10> test_data_3d;
+  ekat::util::md_array<Int,10>        test_index_1d;
+  ekat::util::md_array<Int, 5,10>     test_index_2d;
+  ekat::util::md_array<Int, 2, 5,10>  test_index_3d;
+  Real pi = 2*acos(0.0);
+
+  for (decltype(x_data)::size_type ii=0;ii<x_data.size();ii++) {
+    x_data[ii] = 2.0*pi/x_data.size()*(ii+1);
+  }
+  for (int jj=0;jj<5;jj++) {
+    y_data[jj] = 4.0*pi/y_data.size()*(jj+1);
+  }
+  for (int kk=0;kk<2;kk++) {
+    z_data[kk] = 100*(kk+1);
+  }
+  // Write dimension data and initial fields
+  grid_write_data_array(outfilename,"x",xdim,ekat::util::data(x_data));
+  grid_write_data_array(outfilename,"y",ydim,ekat::util::data(y_data));
+  grid_write_data_array(outfilename,"z",zdim,ekat::util::data(z_data));
+  sync_outfile(outfilename); 
+  // write multiple timesteps of data for comparison:
+  Real dt = 1.0;
+  for (int tt=0;tt<3;tt++) {
+    for (decltype(x_data)::size_type ii=0;ii<x_data.size();ii++) {
+      test_data_1d[ii]  = 0.1 * cos(x_data[ii] + tt*dt); // phase shift by dt
+      test_index_1d[ii] = 10000*tt + ii;
+      for (int jj=0;jj<5;jj++) {
+        test_data_2d[jj][ii]  = test_data_1d[ii] * sin(y_data[jj] + tt*dt); //phase shift by dt
+        test_index_2d[jj][ii] = test_index_1d[ii] + 100*jj;
+        for (int kk=0;kk<2;kk++) {
+          test_data_3d[kk][jj][ii]  = test_data_2d[jj][ii] + z_data[kk];
+          test_index_3d[kk][jj][ii] = test_index_2d[jj][ii] + 1000*kk;
+        } //kk
+      } //jj
+    } //ii
+    pio_update_time(outfilename,tt*dt);
+    grid_write_data_array(outfilename,"index_1d",dimlen_1d,ekat::util::data(test_index_1d));
+    grid_write_data_array(outfilename,"index_2d",dimlen_2d,ekat::util::data(test_index_2d));
+    grid_write_data_array(outfilename,"index_3d",dimlen_3d,ekat::util::data(test_index_3d));
+    grid_write_data_array(outfilename,"data_1d",dimlen_1d,ekat::util::data(test_data_1d));
+    grid_write_data_array(outfilename,"data_2d",dimlen_2d,ekat::util::data(test_data_2d));
+    grid_write_data_array(outfilename,"data_3d",dimlen_3d,ekat::util::data(test_data_3d));
+    sync_outfile(outfilename); 
+  } //tt
+
+  eam_pio_finalize();
+} // TEST scorpio_interface_output
+
```

Comment:
So do I understand correctly that this isn't really a test in the sense that it writes output but never checks if that output is garbage or even exists, right? The test is really "does this test run without crashing"? And I think it needs to be this way because we don't have the ability to read netcdf within SCREAM yet? So actually checking output in this test will occur in the next PR your issue? It's probably a good idea to add a comment to the beginning of this test saying what it does.

Reply:
Good point. Maybe there is something we could check? Like, I don't know, the file size?

Reply:
BFB testing can be conducted using the baseline netCDF file located in the "data" directory and is copied to the local run directory during the compile.  It does take an extra cprnc (or similar) step, but it is pretty robust for checking all output.

Reply:
It is one of the new files included in this PR.